### PR TITLE
multiprocessing test metatensor

### PR DESCRIPTION
**Pull request by wyli**
Signed-off-by: Wenqi Li <wenqil@nvidia.com>

- Fixes https://github.com/Project-MONAI/MONAI/runs/6521801558?check_suite_focus=true

### Description
may need to register new logic to the pickler
according to https://github.com/pytorch/pytorch/blob/53e0d7a3bae621ef0d07cfd44957736e9149dd8e/torch/_tensor.py#L77-L79

cc @rijobro 

### Status
**Ready**

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [x] New tests added to cover the changes.
- [ ] Integration tests passed locally by running `./runtests.sh -f -u --net --coverage`.
- [x] Quick tests passed locally by running `./runtests.sh --quick --unittests  --disttests`.
- [x] In-line docstrings updated.
- [ ] Documentation updated, tested `make html` command in the `docs/` folder.

### CODE REVIEW

**Comment by rijobro** [comment]
Should these not go into `dev` rather than `feature/MetaTensor`?

File: tests/test_meta_tensor.py [hunk 1]
```diff
@@ -488,6 +491,15 @@ def test_construct_with_pre_applied_transforms(self):
         m = MetaTensor(im, applied_operations=data[PostFix.transforms(key)])
         self.assertEqual(len(m.applied_operations), len(tr.transforms))
 
+    @SkipIfBeforePyTorchVersion((1, 8, 1))
+    def test_multiprocessing(self):
```
Comment (rijobro) [inline tests/test_meta_tensor.py:495]: Is this test alone enough to satisfy ourselves that multi-processing is supported?

Comment (wyli) [reply tests/test_meta_tensor.py:495]: this is mainly to fix the current integration error, I think there might be more issues when we refactor more components, so I think maybe it's fine to merge this to `feature/MetaTensor` first...

**Comment by wyli** [comment]
> Should these not go into `dev` rather than `feature/MetaTensor`?

thanks @rijobro, I think this one is now good for `dev`, it also fixes a constructor issue identified during testing:

```py
>>> from monai.data import MetaTensor
>>> MetaTensor([1.0], device="cpu")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: __init__() got an unexpected keyword argument 'device'

```

File: monai/data/meta_tensor.py [hunk 2]
```diff
@@ -82,10 +83,17 @@ def __new__(
         *args,
         **kwargs,
     ) -> MetaTensor:
-        return torch.as_tensor(x, *args, **kwargs).as_subclass(cls)  # type: ignore
+        _kwargs = {"device": kwargs.pop("device", None), "dtype": kwargs.pop("dtype", None)} if kwargs else {}
```
Comment (rijobro) [inline monai/data/meta_tensor.py:86]: If doing `device`, should we also do `dtype`?

Comment (wyli) [reply monai/data/meta_tensor.py:86]: yes, the current pr supports both `device` and `dtype`

Comment (rijobro) [reply monai/data/meta_tensor.py:86]: Oh sorry I missed that!

**Review by rijobro — APPROVED** [review]

**Pull request merged**